federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -647,13 +648,15 @@ func (ic *IngressController) getMasterCluster() (master *federationapi.Cluster,
   If there is no master cluster, then fallbackUID is used (and hence this cluster becomes the master).
 */
 func (ic *IngressController) updateClusterIngressUIDToMasters(cluster *federationapi.Cluster, fallbackUID string) (string, error) {
-	masterCluster, masterUID, err := ic.getMasterCluster()
 	clusterObj, clusterErr := api.Scheme.DeepCopy(cluster) // Make a clone so that we don't clobber our input param
 	cluster, ok := clusterObj.(*federationapi.Cluster)
 	if clusterErr != nil || !ok {
-		glog.Errorf("Internal error: Failed clone cluster resource while attempting to add master ingress UID annotation (%q = %q) from master cluster %q to cluster %q, will try again later: %v", uidAnnotationKey, masterUID, masterCluster.Name, cluster.Name, err)
-		return "", err
+		errmsg := fmt.Sprintf("Internal error: Failed clone cluster %q resource: %v", cluster.Name, clusterErr)
+		glog.Errorf(errmsg)
```

Comment:
It's not clear to me that it's a good idea to both log the error and return it (although I realize that the original code did).  Surely the caller will log the error if you return it here?  It seems like the correct general strategy is to either return the error, or log it, not both?